gnovm/cmd/gno/test.go

```diff
@@ -211,26 +226,52 @@ func execTest(cfg *testCfg, args []string, io commands.IO) error {
 
 		memPkg := gno.MustReadMemPackage(pkg.Dir, gnoPkgPath)
 
-		startedAt := time.Now()
-		hasError := catchRuntimeError(gnoPkgPath, io.Err(), func() {
-			err = test.Test(memPkg, pkg.Dir, opts)
-		})
+		var syncWrite sync.Mutex
 
-		duration := time.Since(startedAt)
-		dstr := fmtDuration(duration)
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
 
-		if hasError || err != nil {
-			if err != nil {
-				io.ErrPrintfln("%s: test pkg: %v", pkg.Dir, err)
+			if err := sem.Acquire(ctx, 1); err != nil {
```

Comment:
This could be a channel where you write at the beginning `ch <- struct{}{}` and do a `defer func() { <- ch }` at the end.